### PR TITLE
[devbox] Use the devbox install action, update devbox packages.

### PR DESCRIPTION
**Pull request by mdwn**
The devbox install action will be used instead of our custom devbox install. This will ensure caching is used, speeding up this process.

Additionally, the following have been updated:
* buf -> 1.23.1.
* go -> 1.20.5
* clang -> 14.0.6

Finally, packages have been migrated to regular devbox syntax where possible.

### CODE REVIEW

File: devbox.json [hunk 1]
```diff
@@ -2,11 +2,11 @@
...
     "addlicense@1.0.0",
     "bats@1.3.0",
-    "github:nixos/nixpkgs/d7f28652048b3bed6a512542e62ea1a50691e349#buf",
-    "clang@11.1.0",
+    "github:nixos/nixpkgs/2920b6fc16a9ed5d51429e94238b28306ceda79e#buf",
+    "llvmPackages_14.clangUseLLVM@14.0.6",
     "gci@0.9.1",
-    "github:nixos/nixpkgs/deb3d80ae0ccafe4f19d3edf32e2eb7fda283978#go",
-    "github:nixos/nixpkgs/22540da6c19a41512dd6bc575858e1215baeb3f9#golangci-lint",
+    "go@1.20.5",
```
Comment (AntonAM) [inline devbox.json:8]: I think Alan recently bumped go to 1.20.6 everywhere

**Review by AntonAM — APPROVED** [review]

File: devbox.json (hunk 1 above)
Comment (AntonAM) [reply devbox.json:8]: And also buf to 1.24 - https://github.com/gravitational/teleport/pull/29081

**Review by hugoShaka — APPROVED** [review]

**Review by strideynet — APPROVED** [review]

**Review by codingllama — APPROVED** [review]

File: devbox.json (hunk 1 above)
Comment (mdwn) [reply devbox.json:8]: Yep, both go and buf are still lagging here. go in particular usually has a ~2ish week lag time for version bumps.

**Comment by jakule** [comment]
I'm happy with using the latest Clang, except for the prod build, we're using Clang 10.0.1. I don't think this makes a huge difference, but you may be supprised when "it works" in devbox, but it breaks the prod/Ci build.

**Comment by mdwn** [comment]
> I'm happy with using the latest Clang, except for the prod build, we're using Clang 10.0.1. I don't think this makes a huge difference, but you may be supprised when "it works" in devbox, but it breaks the prod/Ci build.

Interestingly, it appears that the clang@10x packages are broken in nixpkgs somehow, so I guess we can't use them regardless.

**Review by r0mant — APPROVED** [review]

**Pull request merged**